crates/ruff_linter/src/rules/flake8_implicit_str_concat/rules/explicit.rs

```diff
@@ -35,45 +39,133 @@ use crate::settings::LinterSettings;
 pub(crate) struct ExplicitStringConcatenation;
 
 impl Violation for ExplicitStringConcatenation {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Always;
+
     #[derive_message_formats]
     fn message(&self) -> String {
         "Explicitly concatenated string should be implicitly concatenated".to_string()
     }
+
+    fn fix_title(&self) -> Option<String> {
+        Some("Remove redundant '+' operator to implicitly concatenate".to_string())
+    }
 }
 
 /// ISC003
-pub(crate) fn explicit(
-    expr: &Expr,
-    locator: &Locator,
-    settings: &LinterSettings,
-) -> Option<Diagnostic> {
+pub(crate) fn explicit(expr: &Expr, checker: &Checker) -> Option<Diagnostic> {
     // If the user sets `allow-multiline` to `false`, then we should allow explicitly concatenated
     // strings that span multiple lines even if this rule is enabled. Otherwise, there's no way
     // for the user to write multiline strings, and that setting is "more explicit" than this rule
     // being enabled.
-    if !settings.flake8_implicit_str_concat.allow_multiline {
+    if !checker.settings.flake8_implicit_str_concat.allow_multiline {
         return None;
     }
 
-    if let Expr::BinOp(ast::ExprBinOp {
-        left,
-        op,
-        right,
-        range,
-    }) = expr
-    {
+    if let Expr::BinOp(bin_op) = expr {
+        let ast::ExprBinOp {
+            left,
+            right,
+            range,
+            op,
+        } = bin_op;
         if matches!(op, Operator::Add) {
             if matches!(
                 left.as_ref(),
                 Expr::FString(_) | Expr::StringLiteral(_) | Expr::BytesLiteral(_)
             ) && matches!(
                 right.as_ref(),
                 Expr::FString(_) | Expr::StringLiteral(_) | Expr::BytesLiteral(_)
-            ) && locator.contains_line_break(*range)
+            ) && checker.locator().contains_line_break(*range)
             {
-                return Some(Diagnostic::new(ExplicitStringConcatenation, expr.range()));
+                let mut diagnostic = Diagnostic::new(ExplicitStringConcatenation, expr.range());
+                diagnostic.try_set_fix(|| generate_fix(checker, bin_op));
```

Comment:
Given that this uses `try_set_fix`, this would suggest that this fix isn't always available. 

This makes me wonder. Should, instead, the rule be changed that it doesn't flag cases where the left and right can't be implicitly concatenated?